google/cloud/documentai_toolbox/wrappers/entity.py

```diff
@@ -57,28 +57,33 @@ class Entity:
     page_offset: dataclasses.InitVar[Optional[int]] = 0
 
     type_: str = dataclasses.field(init=False)
-    mention_text: str = dataclasses.field(init=False, default="")
-    normalized_text: str = dataclasses.field(init=False, default="")
+    mention_text: Optional[str] = dataclasses.field(init=False, default=None)
+    normalized_text: Optional[str] = dataclasses.field(init=False, default=None)
 
-    start_page: int = dataclasses.field(init=False)
-    # Only Populated for Splitter/Classifier Output
-    end_page: int = dataclasses.field(init=False)
+    # Not populated in Classifier output
+    start_page: Optional[int] = dataclasses.field(init=False, default=None)
+    # Only Populated for Splitter Output
```

Comment:
Some of these conditions become quite subtle.  Should each Document remember if it came from a Splitter or Classifier or...?  We could add some metadata for it if it helps validation.

Reply:
They do become subtle. I don't think the Document needs to know explicitly what type of processor it came from. This isn't output anywhere in the Document.JSON